src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -4165,19 +4174,15 @@ def _structure_accessor_handler(self, node, parent):
             top_ref = node
             member_nodes = []
 
-        # If we encounter array ranges while processing this derived-type
-        # access then we will need the symbol being referred to so we create
-        # that first.
         if isinstance(top_ref, Fortran2003.Name):
+            # Update the top_ref symbol information knowing that it exits
```

Comment:
I'm not quite sure what this comment (and the one below) means? Are you saying that because it's the 'root' of the structure access it must appear in a symbol table (rather than saying that it must already be present)?

Reply:
Oh yeah, updated to  "Add the structure root reference to the symbol table if its not already there."